libraries/AP_RCProtocol/AP_RCProtocol_DroneCAN.cpp

```diff
@@ -90,12 +90,38 @@ void AP_RCProtocol_DroneCAN::update()
         }
         last_receive_ms = rcin.last_sample_time_ms;
 
+        if (rcin.bits.QUALITY_VALID) {
+            frontend._rc_link_status.rssi = rcin.quality;
+        } else
+        if (rcin.bits.QUALITY_LQ) {
```

Comment:
```suggestion
        switch (QualityType)rcin.bits.QUALITY_TYPE)) {
        case QualityType::RSSI:  // this would be value zero
```

similarly elsewhere

Reply:
understood